addons/crm/models/crm_lead.py

```diff
@@ -818,7 +818,22 @@ def get_rainbowman_message(self):
 
     def _get_rainbowman_message(self):
         message = False
-        if self.user_id and self.team_id and self.expected_revenue:
+        if self.user_id and self.team_id:
+            if not self.expected_revenue:
+                # Show rainbow man for the first won lead of a salesman, even if expected revenue is not set. It is not
+                # very often that leads without revenues are marked won, so simply get count using ORM instead of query
+                today = fields.Datetime.today()
+                user_won_leads_count = self.search_count([
+                    ('type', '=', 'opportunity'),
+                    ('user_id', '=', self.user_id.id),
+                    ('probability', '=', 100),
+                    ('date_closed', '>=', date_utils.start_of(today, 'year')),
+                    ('date_closed', '<', date_utils.end_of(today, 'year')),
+                ])
+                if user_won_leads_count == 1:
+                    message = _('Go, go, go! Congrats for your first deal.')
+                return message
```

Comment:
I bet you will have some issues if user_won_leads_count is not 1.

Reply:
Not sure what issues. Can you explain ?

Reply:
Sorry, message is defined above. Strange construct actually I was a bit lost.

Maybe you could make some "sections" to ease understanding, like
```
if not user_id or not team_id:
    return False
if not expected_revenue:
     # compute stuff wo revenue
     if len(leads) == 1:
        message = ...
        return message
    return False
# compute stuff with revenue
message = ...
return message
```